### ISSUE TITLE
Не работает автоматическая сервисная печать заказа после его оплаты через API V7 Preview6 iiko 7.7.6020

### ISSUE THREAD

**Issue by onestepchik** [iiko/front.api.sdk#249]
Коллеги, нужна помощь. У нас не работает автоматическая сервисная печать заказа после оплаты.
iiko 7.7.6020
API V7 Preview6

Создаем плагином заказ и вызываем его оплату.
PluginContext.Operations.AddExternalPaymentItem(order.ResultSum, false, additionalData, null, paymentType, order, cred);

В настройках ТП стоит галка автоматическая сервисная печать после оплаты заказа.
После оплаты заказа сервисная печать не вызывается.

Логи приложили.
Время заказа 27.07.2021 18:03
ID заказа a0980a85-d1dd-46f3-9077-924855fa3638

Скрин настроек группы
<img width="774" alt="Настройки группы" src="https://user-images.githubusercontent.com/85111425/127143218-53c5c2e4-457a-440e-ad0c-4fa8b110fb9e.png">

Скрин настроек отделения
<img width="1046" alt="Настройки отделения" src="https://user-images.githubusercontent.com/85111425/127143231-b5133187-98c9-4778-a255-21cefe0188d2.png">

[agent.2021-07-26.log](https://github.com/
[cash-server.2021-07-26.log](https://github.com/iiko/front.api.sdk/files/6884830/cash-server.2021-07-26.log)
[preliminary-orders.2021-07-26.log](https://github.com/iiko/front.api.sdk/files/6884831/preliminary-orders.2021-07-26.log)
[virtual-printer.log](https://github.com/iiko/front.api.sdk/files/6884832/virtual-printer.log)
iiko/front.api.sdk/files/6884827/agent.2021-07-26.log)
[api.2021-07-26.log](https://github.com/iiko/front.api.sdk/files/6884829/api.2021-07-26.log)

**Comment by onestepchik** [comment]
Коллеги, поясню, почему важно, чтобы была возможность поддержки автоматической сервисной печати после оплаты заказа. В нашем кейсе оплата заказа происходит полностью на стороне iiko и отправлять блюда на печать нужно только при успешной оплате. Мы попробовали отправить команду печати блюд PrintOrderItems после метода Pay, в результате получаем ошибку, что нельзя изменять заказ с статусе CLOSED. Заказ после оплаты становится закрытым. Поэтому технически мы не можем вызвать сервисную печать вручную. До оплаты вызывать печать блюд в нашем сценарии ПО киоска нельзя, так как тогда могут быть заказы, которые покупатели по факту не смогут оплатить, а блюда начнут готовиться из-за заранее отправленных заявок на приготовление блюд по цехам. Мы расчитывали на опцию в отделении "Автоматическая сервисная печать после оплаты", но к сожалению автоматически печать не происходит, а она очень была бы кстати. Я так понимаю, та же ситуация будет в сценарии проброса заказа в стол с оплатой и последующим закрытием через iikoTransport. Очень ждем решения по сложившейся ситуации. Спасибо!

**Comment by onestepchik** [comment]
<img width="520" alt="Снимок экрана 2021-07-27 в 15 35 49" src="https://user-images.githubusercontent.com/85111425/127167437-7126adbb-4d4e-4613-a65d-5a94f3c821af.png">
Настройка, на которую мы делали ставку ;)

**Comment by vcpp** [comment]
> У нас не работает автоматическая сервисная печать заказа после оплаты.

Простите, не у вас, а у нас :-)
Это наша недоделка, уже который год висит в планах, но руки не доходят. Подняли приоритет, постараемся доделать в следующей версии (7.8). Внутренний номер: RMS-50939.

**Comment by onestepchik** [comment]
Доброго дня! Спасибо большое! Мы если что готовы помочь в тестировании, так как задача очень актуальна для нас и не только. Очень будем ждать доработки данной задачи, если необходимо какая-либо информация, будем рады сотрудничать. (+79212629050 Остап для оперативной связи)

**Comment by onestepchik** [comment]
> > У нас не работает автоматическая сервисная печать заказа после оплаты.
> 
> Простите, не у вас, а у нас :-)
> Это наша недоделка, уже который год висит в планах, но руки не доходят. Подняли приоритет, постараемся доделать в следующей версии (7.8). Внутренний номер: RMS-50939.

Коллеги, мы же правильно понимаем, что после фикса, заработает автоматическая сервисная печать. В нашем случае киоска, мы добавляем тип оплаты потом вызываем его оплату. 
Пример кода            
PluginContext.Operations.AddExternalPaymentItem(order.ResultSum, false, additionalData, null, paymentType, order, cred);
PluginContext.Operations.PayOrder(cred, order);

**Comment by vcpp** [comment]
> Мы же правильно понимаем, что после фикса, заработает автоматическая сервисная печать после PayOrder, если стоит соответствующая настройка в iikoOffice?

Да, правильно. Автоматическая сервисная печать будет срабатывать до или после оплаты в соответствии с настройкой при вызове метода `PayOrder` для всех поддерживаемых версий API. Доработки в плагине не понадобятся, будет достаточно обновить iikoFront (iikoRMS). Постараемся успеть к релизу 7.8.
